tensorflow/core/util/mkl_util.h

```diff
@@ -1853,13 +1893,59 @@ class MklPrimitiveFactory {
   ~MklPrimitiveFactory() {}
 
   MklPrimitive* GetOp(const string& key) {
+#ifndef DNNL_AARCH64_USE_ACL
     auto& lru_cache = MklPrimitiveFactory<T>::GetLRUCache();
     return lru_cache.GetOp(key);
+#else
+    while(true) {
+      mutex_lock lock(mtx_);
+      auto& lru_cache = MklPrimitiveFactory<T>::GetLRUCache();
+
+      // check to see whether primitive already exists
+      MklPrimitive *primitive = lru_cache.GetOp(key);
+      if(primitive != nullptr) {
+        return primitive;
+      }
+
+      // now check whether some other thread is
+      // creating this primitive
+      if(!lru_cache.IsAllocating(key)) {
+        // this thread is going to pick it up and
+        // and create the primitive
+        lru_cache.Allocate(key);
+        return nullptr;
+        // now we release lock
+        // as primitive creating might take long time
+      }
+
+      // at this point we cannot create primitive
+      // as someone else is creating so we
+      // should wait for primitive to get created and
+      // then return created primitive
+      cv_.wait(lock);
```

Comment:
I'm a bit worried about this part since it's quite loose and assumes a few things. But I'm okay with keeping it like this for now (TF 2.9).

Reply:
I have added hopefully clearer explanation in ecb7614, but please do let me know if something else should be done.